compute environment sha unique

evgeniilezhnin opened the pull request.

conradoverta committed:
Checkpoint

conradoverta committed:
Compiles

conradoverta committed:
Generate from proto

conradoverta committed:
Add proto generation

conradoverta committed:
Add diff

conradoverta committed:
Merge branch 'master' into cm/generate

conradoverta committed:
Create diff computer

conradoverta committed:
Add documentation to computeListDiff

conradoverta committed:
Add diff merger

conradoverta committed:
Add type checker

conradoverta committed:
Fix string comparison

conradoverta committed:
Handle cases where things match

1. Add `equal` and `isEmpty` methods
2. If a diff is empty, return null
3. If two objects are the same, the diff is null

conradoverta committed:
Fix equals method

conradoverta committed:
Fix mergeNotebookCode

Commit:
correction in protos

conradoverta committed:
Fix merger

conradoverta committed:
Update templates

conradoverta committed:
Update protos and generated files

conradoverta committed:
Merge branch 'cm/generate' of github.com:VertaAI/modeldb into cm/generate

conradoverta committed:
Remove empty when loading from proto

conradoverta committed:
Fix equals

conradoverta committed:
Add hash code

conradoverta committed:
Add override

evgeniilezhnin committed:
clear version

evgeniilezhnin committed:
don't use list when compute hash

anandjakhaniya commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -208,7 +208,8 @@ private void validateEnvironmentVariableName(String name) throws ModelDBExceptio
     sb.append("env:");
-    for (EnvironmentVariablesBlob environmentVariablesBlob : blob.getEnvironmentVariablesList()) {
+    for (EnvironmentVariablesBlob environmentVariablesBlob :
+        new HashSet<>(blob.getEnvironmentVariablesList())) {
```
Why this change required? it just for loop you can use whatever I mean List or Set I think it doesn't affect to hash also.

evgeniilezhnin replied:
HashSet ignores order of elements. That's why I have used it. But it doesn't guarantee order, so I better change it to sort.

conradoverta committed:
Run formatter

conradoverta committed:
Use Optional in diff computation

conradoverta committed:
Fix getStatus

conradoverta committed:
Fixing merger to use optional

conradoverta committed:
Remove getOrNull

conradoverta committed:
Revert "Remove getOrNull"

This reverts commit 57f161c7ff0141d5062b457080ff291f78336ee7.

conradoverta committed:
Revert "Fixing merger to use optional"

This reverts commit 4b9c337b32fb028cf568b09956f3bfff7a449e74.

conradoverta committed:
Revert "Fix getStatus"

This reverts commit b3215f3efd2be136a22a6ef035e14a5190e347b8.

conradoverta committed:
Revert "Use Optional in diff computation"

This reverts commit 210e813dac9fd871def69d2c940fb14977a13120.

conradoverta committed:
Fix protobuf for blobs

conradoverta committed:
Update the API so that status is associated with one end entity

conradoverta committed:
Add validation for proto conversion of diffs

conradoverta committed:
Diff and merge tests pass sort of

The only reason they don't pass right now is because the lists might end
up in a different order.

conradoverta committed:
Fix diff and merge tests

We do a dumb merge right now that overwrites everything. We can define a
smarter merger and pass as argument as desired.

We should also put the string sorting inside the autogenerated code, I
think, instead of outside. It will help be consistent all accross. We
can also push the hash generation to the autogenerated code.

ravishetye commented:
@evgeniilezhnin  this can be closed? or still, needs review?

conradoverta committed:
Started adding getSHA and sort properties by name

conradoverta committed:
Update scala client

evgeniilezhnin commented:
> @evgeniilezhnin this can be closed? or still, needs review?

it needs review

evgeniilezhnin committed:
VR-3656 compute set hash

evgeniilezhnin committed:
fix

ravishetye commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -227,23 +222,9 @@ private String computeSHA(PythonEnvironmentBlob blob) throws NoSuchAlgorithmExce
-          .append(pythonRequirementEnvironmentBlob.getConstraint());
-      appendVersion(sb, pythonRequirementEnvironmentBlob.getVersion());
-    }
+    sb.append(new HashSet<>(blob.getRequirementsList()).hashCode());
```
would hahsCode be same across calls/ across restarts?
we would want the same content of requirementList to be create identical sha

evgeniilezhnin replied:
changed base and used generated files to compute hash

evgeniilezhnin committed:
Feature/vr 3656 blob diff refactoring with tests (#360)

* VR-3656 blob DAO fixes

* VR-3656 list merge fixes

* VR-3656 diff status fix

* VR-3656 small fix

conradoverta committed:
Update API as per Kirill's request (#365)

Now, anything that has `status`, `A` and `B` only have those specific
elements.

conradoverta committed:
Update merge protos (#372)

conradoverta commented:
The autogenerated code has hash computation, which should be stable (it checks for new fields so that we can add them without changing the SHA). If we want to change how SHA is computed, should we use that instead?

evgeniilezhnin committed:
Merge branch 'feature/diffapi' into feature/clear_version

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/diffFactory/DatasetBlobDiffFactory.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/diffFactory/EnvironmentBlobDiffFactory.java
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningEnvironmentVariablesDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningHyperparameterConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningHyperparameterSetConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningNotebookCodeBlob.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningNotebookCodeDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPathDatasetComponentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPythonEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPythonRequirementEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningS3DatasetComponentDiff.scala
#	client/verta/verta/_protos/public/modeldb/versioning/Config_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Environment_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/go/protos/public/modeldb/versioning/Config.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/Environment.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.go
#	protos/gen/python/protos/public/modeldb/versioning/Config_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Environment_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/swagger/protos/public/modeldb/versioning/VersioningService.swagger.json
#	protos/protos/public/modeldb/versioning/Config.proto
#	protos/protos/public/modeldb/versioning/Environment.proto

evgeniilezhnin committed:
use generated files

evgeniilezhnin committed:
use generated files

evgeniilezhnin committed:
format

Commit:
protos change for diff to be with common parent

conradoverta committed:
Enforce that lists inside autogenerated are sorted (#369)

* Enforce that lists inside autogenerated are sorted

* Only sort if autogenerated type

evgeniilezhnin committed:
Merge branch 'master' into feature/diffapi

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/container/CodeContainer.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/diffFactory/DatasetBlobDiffFactory.java
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/api/VersioningServiceApi.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbArtifact.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbCodeVersion.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbExperimentRun.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbGitSnapshot.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbObservation.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbVersioningEntry.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningEnvironmentVariablesDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningHyperparameterConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningHyperparameterSetConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningListBlobExperimentRunsRequestResponse.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningListCommitExperimentRunsRequestResponse.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningMergeRepositoryCommitsRequest.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningNotebookCodeBlob.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningNotebookCodeDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPathDatasetComponentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPythonEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPythonRequirementEnvironmentDiff.scala
#	client/verta/verta/_protos/public/modeldb/versioning/Config_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Environment_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/go/protos/public/modeldb/versioning/Config.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/Environment.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.gw.go
#	protos/gen/python/protos/public/modeldb/versioning/Config_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Environment_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/swagger/protos/public/modeldb/versioning/VersioningService.swagger.json
#	protos/protos/public/modeldb/versioning/Config.proto
#	protos/protos/public/modeldb/versioning/Environment.proto

evgeniilezhnin committed:
Merge branch 'master' into feature/diffapi

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/container/CodeContainer.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/diffFactory/DatasetBlobDiffFactory.java
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/api/VersioningServiceApi.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbArtifact.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbCodeVersion.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbExperimentRun.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbGitSnapshot.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbObservation.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbVersioningEntry.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningEnvironmentVariablesDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningHyperparameterConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningHyperparameterSetConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningListBlobExperimentRunsRequestResponse.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningListCommitExperimentRunsRequestResponse.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningMergeRepositoryCommitsRequest.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningNotebookCodeBlob.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningNotebookCodeDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPathDatasetComponentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPythonEnvironmentDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningPythonRequirementEnvironmentDiff.scala
#	client/verta/verta/_protos/public/modeldb/versioning/Config_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Environment_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/go/protos/public/modeldb/versioning/Config.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/Environment.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.gw.go
#	protos/gen/python/protos/public/modeldb/versioning/Config_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Environment_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/swagger/protos/public/modeldb/versioning/VersioningService.swagger.json
#	protos/protos/public/modeldb/versioning/Config.proto
#	protos/protos/public/modeldb/versioning/Environment.proto

anandjakhaniya commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -208,12 +208,15 @@ private void validateEnvironmentVariableName(String name) throws ModelDBExceptio
+                .EnvironmentBlob.fromProto(blob)
+                .EnvironmentVariables;
+    sb.append(
+        environmentVariables == null ? "null" : new HashSet<>(environmentVariables).hashCode());
```
why you are set to null as a string in StringBuilder? here something wrong. you should just leave it if it is null.

evgeniilezhnin replied:
fixed

anandjakhaniya commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -226,24 +229,15 @@ private String computeSHA(PythonEnvironmentBlob blob) throws NoSuchAlgorithmExce
-      appendVersion(sb, pythonRequirementEnvironmentBlob.getVersion());
-    }
+    sb.append(
+        blobNew.Requirements == null ? "null" : new HashSet<>(blobNew.Requirements).hashCode());
```
Same comment like above

evgeniilezhnin replied:
fixed

anandjakhaniya commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -226,24 +229,15 @@ private String computeSHA(PythonEnvironmentBlob blob) throws NoSuchAlgorithmExce
-          .append(pythonConstraintEnvironmentBlob.getConstraint());
-      appendVersion(sb, pythonConstraintEnvironmentBlob.getVersion());
-    }
+    sb.append(blobNew.Constraints == null ? "null" : new HashSet<>(blobNew.Constraints).hashCode());
```
same as above comments

evgeniilezhnin replied:
fixed

evgeniilezhnin committed:
Merge branch 'master' into feature/diffapi

Commit:
Merge branch 'master' into feature/diffapi

Commit:
Merge branch 'feature/diffapi' of github.com:VertaAI/modeldb into feature/diffapi

evgeniilezhnin committed:
Merge branch 'feature/diffapi' into feature/clear_version

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/versioning/BlobDAO.java
#	backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningS3DatasetComponentDiff.scala
#	client/verta/verta/_protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.go
#	protos/gen/python/protos/public/modeldb/versioning/VersioningService_pb2.py
#	webapp/client/src/core/shared/view/domain/TagsManager/Tag/Tag.tsx
#	webapp/client/yarn.lock

evgeniilezhnin committed:
pr fixes

evgeniilezhnin committed:
pr fixes

evgeniilezhnin committed:
Merge branch 'master' into feature/diffapi

evgeniilezhnin committed:
regenerate

evgeniilezhnin committed:
Merge branch 'master' into feature/diffapi

# Conflicts:
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/api/VersioningServiceApi.scala
#	client/verta/verta/_protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.gw.go
#	protos/gen/python/protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/swagger/protos/public/modeldb/versioning/VersioningService.swagger.json

evgeniilezhnin committed:
regenerate

evgeniilezhnin committed:
Vr 3616 fix diff and add commit from diff bugs (#376)

* VR-3616 fix diff and add commit from diff bug

conradoverta committed:
Change name of autogenerated entities (#409)

* Change name of autogenerated entities

* change util names (#410)

Co-authored-by: evgeniilezhnin <eugene.lezhnin@quantumsoft.ru>

evgeniilezhnin committed:
fix

Commit:
merge on backend (#373)

* VR-3616 diff test start

* VR-3656 blob DAO fixes

* VR-3656 list merge fixes

* VR-3656 diff status fix

* VR-3656 rebased

* protos for merge api v1 (#368)

* protos for merge api v1

* restful

* review feedback

* plumbing

* merge commit v1

* review feedback

* VR-3616 add a test case

* VR-3616 fix add commit from diff bugs

* make work

* VR-3703

* VR-3616 fixes

* VR-3616 format

* Compare using the commit hash

* VR-3616 get generated files

* VR-3616 checks

* VR-3616 checks

* correctly use commit hash instead of rootSHA

* VR-3616 keep only fixes without validation

* VR-3616 comments

* VR-3616 use linked set

* break session into two

* protos gen

* Revert "List commits log API (#383)"

This reverts commit a65ddcb1739fc90ebd39d70c7362152166f7fd98.

* Revert "Revert "List commits log API (#383)""

This reverts commit 3f96850c0fa9aa1e290629c365de6abf85114971.

* remove stray change

* random changes removed

Co-authored-by: Evgenii Lezhnin <eugene.lezhnin@quantumsoft.ru>
Co-authored-by: Conrado Miranda <conrado@verta.ai>

ravishetye commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -220,12 +220,16 @@ private void validateEnvironmentVariableName(String name) throws ModelDBExceptio
+                .getEnvironmentVariables();
+    if (environmentVariables != null) {
+      sb.append("env:");
+      sb.append(new HashSet<>(environmentVariables).hashCode());
```
Are you sure Hash set will guarantee you the same order across restarts? We want the same set of envVars to map to same SHA.

Why not use environmentVariables itself? It is getting sorted in the auto gen code.

evgeniilezhnin replied:
oh sorry. Forgot about that. Fixed it. Also added a small change into template to use setters instead of set operator to force sort in fromProto.

ravishetye commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -239,23 +243,21 @@ private String computeSHA(PythonEnvironmentBlob blob) throws NoSuchAlgorithmExce
+        .getRequirements();
+    if (requirements != null) {
+      sb.append(":requirements:");
+      sb.append(new HashSet<>(requirements).hashCode());
```
similar to above.

evgeniilezhnin replied:
fixed

ravishetye replied:
shouldn't this be removed?

ravishetye commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java:
```diff
@@ -239,23 +243,21 @@ private String computeSHA(PythonEnvironmentBlob blob) throws NoSuchAlgorithmExce
+        .getConstraints();
+    if (constraints != null) {
+      sb.append(":constraints:");
+      sb.append(new HashSet<>(constraints).hashCode());
```
same.

evgeniilezhnin replied:
fixed

ravishetye reviewed (commented):
I do not get the requirement of hashSet here

evgeniilezhnin committed:
Merge branch 'feature/diffapi' into feature/clear_version

evgeniilezhnin committed:
fix

evgeniilezhnin committed:
fix

evgeniilezhnin committed:
Merge branch 'feature/diffapi' into feature/clear_version

evgeniilezhnin committed:
VR-3616 test cases (#377)

* VR-3616 diff test cases

evgeniilezhnin committed:
Merge branch 'feature/diffapi' into feature/clear_version

ravishetye commented on backend/src/main/java/ai/verta/modeldb/versioning/blob/visitors/Visitor.java:
```diff
@@ -3,101 +3,103 @@
     if (lst == null) return;
-    for (AutogenBlob val : lst) {
-      preVisitAutogenBlob(val);
+    for (AutogenBlobDiff val : lst) {
```
I do not understand why this file changed

conradoverta replied:
I believe this file is auto-generated (it might be missing the header).

evgeniilezhnin replied:
for some reason I have different result than in master when I run ./swagger_codegen.sh and reformat step in maven. Did I miss something here?

evgeniilezhnin replied:
It mostly imports and method order.

I did just a small change in generation template. But when I don't do any changes it's still different

ravishetye replied:
@conradoverta : could you review the autogen files and the process. the code merged from #425 makes sense to me otherwise.

ravishetye commented on backend/src/test/java/ai/verta/modeldb/blobs/BlobEquality.java:
```diff
@@ -1,22 +1,29 @@
-import static org.junit.Assert.*;
-import static org.junit.Assume.*;
-
+import ai.verta.modeldb.ModelDBException;
```
i do not understand why this file changed

evgeniilezhnin committed:
Merge branch 'master' into feature/clear_version

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenBlobDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenBlobDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenCodeBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenCodeBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenCodeDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenCodeDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenCommandLineEnvironmentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenCommandLineEnvironmentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenConfigBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenConfigBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenConfigDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenConfigDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenContinuousHyperparameterSetConfigBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenContinuousHyperparameterSetConfigBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDatasetBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDatasetBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDatasetDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDatasetDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDiffStatusEnumDiffStatus.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDiffStatusEnumDiffStatusGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDiscreteHyperparameterSetConfigBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDiscreteHyperparameterSetConfigBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDockerEnvironmentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDockerEnvironmentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDockerEnvironmentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenDockerEnvironmentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentVariablesBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentVariablesBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentVariablesDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenEnvironmentVariablesDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenGitCodeBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenGitCodeBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenGitCodeDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenGitCodeDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterConfigBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterConfigBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterConfigDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterConfigDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterSetConfigBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterSetConfigBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterSetConfigDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterSetConfigDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterValuesConfigBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenHyperparameterValuesConfigBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenNotebookCodeBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenNotebookCodeBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenNotebookCodeDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenNotebookCodeDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetComponentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetComponentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetComponentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetComponentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPathDatasetDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonEnvironmentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonEnvironmentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonEnvironmentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonEnvironmentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonRequirementEnvironmentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonRequirementEnvironmentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonRequirementEnvironmentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenPythonRequirementEnvironmentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetComponentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetComponentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetComponentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetComponentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenS3DatasetDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenVersionEnvironmentBlob.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenVersionEnvironmentBlobGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenVersionEnvironmentDiff.java
#	backend/src/main/java/ai/verta/modeldb/versioning/autogenerated/_public/modeldb/versioning/model/AutogenVersionEnvironmentDiffGen.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java
#	backend/src/main/java/ai/verta/modeldb/versioning/blob/visitors/Visitor.java
#	backend/src/test/java/ai/verta/modeldb/blobs/BlobEquality.java
#	backend/src/test/java/ai/verta/modeldb/blobs/BlobProtoEquality.java
#	backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java
#	backend/templates/model.java
#	client/scala/src/main/scala/ai/verta/swagger/_public/artifactstore/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/client/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/common/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/metadata/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/model/ModeldbKeyValueQuery.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbArtifact.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbExperimentRun.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbGitSnapshot.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/ModeldbVersioningEntry.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningConfigDiff.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningListBlobExperimentRunsRequestResponse.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/modeldb/versioning/model/VersioningListCommitExperimentRunsRequestResponse.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/uac/model/RuntimeError.scala
#	client/scala/src/main/scala/ai/verta/swagger/_public/uac/model/UacGetCollaboratorResponse.scala
#	client/verta/verta/_protos/public/modeldb/ExperimentRunService_pb2.py
#	client/verta/verta/_protos/public/modeldb/HydratedService_pb2.py
#	client/verta/verta/_protos/public/modeldb/Job_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Code_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Config_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Dataset_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/Environment_pb2.py
#	client/verta/verta/_protos/public/modeldb/versioning/VersioningService_pb2.py
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.go
#	protos/gen/go/protos/public/modeldb/versioning/VersioningService.pb.gw.go
#	protos/gen/python/protos/public/modeldb/ExperimentRunService_pb2.py
#	protos/gen/python/protos/public/modeldb/HydratedService_pb2.py
#	protos/gen/python/protos/public/modeldb/Job_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Code_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Config_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Dataset_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/Environment_pb2.py
#	protos/gen/python/protos/public/modeldb/versioning/VersioningService_pb2.py

evgeniilezhnin committed:
generated

evgeniilezhnin committed:
generated

evgeniilezhnin committed:
Feature/compute dataset component sha unique (#425)

* compute dataset sha unique

conradoverta committed:
Re-generate files
